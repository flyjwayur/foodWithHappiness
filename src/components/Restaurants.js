import React, { useState } from "react";
import { Blurhash } from "react-blurhash";
import data from "../data/restaurants.json";
import { sortByName } from "../util/sortArray";

const Restaurants = () => {
  const [ascend, setAscend] = useState(true);
  //Arrange restaurants in ascending order initially
  const [sortedRestaurants, setSortedRestaurants] = useState(
    sortByName(data.restaurants, ascend)
  );
  const [isLoading, setIsLoading] = useState([
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true
  ]);

  const finishLoading = idx => {
    let loadingStatus = [...isLoading];
    loadingStatus[idx] = false;
    setIsLoading(loadingStatus);
  };

  return (
    <div>
      <button
        onClick={() => {
          setAscend(!ascend);
          setSortedRestaurants(sortByName(data.restaurants, ascend));
        }}
      >
        <span>Ascending/Descending order</span>
      </button>
      {sortedRestaurants.map((restaurant, i) => {
        return (
          <div key={restaurant.name + "_" + i}>
            <div>{restaurant.name}</div>
            <div>{restaurant.description}</div>
            <div>{restaurant.delivery_price}</div>
            <Blurhash
              style={{ display: isLoading[i] ? "inline-block" : "none" }}
              hash={restaurant.blurhash}
              width={400}
              height={300}
              resolutionX={32}
              resolutionY={32}
              punch={1}
            />
            <img
              style={{ display: isLoading[i] ? "none" : "inline-block" }}
              src={restaurant.image}
              alt={restaurant.name + "_image"}
              width="400px"
              height="300px"
              onLoad={() => finishLoading(i)}
            />
          </div>
        );
      })}
    </div>
  );
};

export default Restaurants;
