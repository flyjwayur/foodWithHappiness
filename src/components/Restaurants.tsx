import React, { useState, FunctionComponent } from "react";
import { IDataRestaurant, IPropRestaurant } from "../index.d"
import Restaurant from './Restaurant';

const Restaurants: FunctionComponent<{ restaurantsList: IDataRestaurant[], isContentsLoading: boolean }> =
  ({ restaurantsList, isContentsLoading }) => {

    return (
      <div>
        {restaurantsList.map((restaurant: IDataRestaurant, i: number) => {
          const { name, description, delivery_price, blurhash, image } = restaurant;
          const restaurantInfo = { name, description, delivery_price, blurhash, image }
          return (
            <Restaurant
              key={restaurant.name + "_" + i}
              restaurant={restaurantInfo}
              isContentsLoading={isContentsLoading}
            />
          );
        })}
      </div>
    );
  };

export default Restaurants;
