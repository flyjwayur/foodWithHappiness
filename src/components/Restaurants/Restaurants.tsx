import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import { IDataRestaurant } from '../../index.d';
import Restaurant from '../Restaurant/Restaurant';
import { mediaQuerySizes } from '../../theme/mediaQuery';

const VenueCardContainer = styled.div`
  display: flex;
  justify-content: flex-start;
  margin: 0 -0.5em;
  flex-wrap: wrap;

  > div {
    width: calc(33% - 1.065rem);
    margin: 0 0.65rem;
    margin-bottom: 1rem;
  }

  @media (max-width: ${mediaQuerySizes.tabletLandscape}) {
    justify-content: center;
  }

  @media (max-width: ${mediaQuerySizes.phone}) {
    > div {
      width: 100%;
    }
  }
`;

const Restaurants: FunctionComponent<{
  restaurantsList: IDataRestaurant[];
  isContentsLoading: boolean;
}> = ({ restaurantsList, isContentsLoading }) => {
  return (
    <VenueCardContainer>
      {restaurantsList.map((restaurant: IDataRestaurant, i: number) => {
        const {
          name,
          description,
          delivery_price,
          blurhash,
          image,
          tags,
          online,
          currency,
          city
        } = restaurant;
        const restaurantInfo = {
          name,
          description,
          delivery_price,
          blurhash,
          image,
          tags,
          online,
          currency,
          city
        };
        return (
          <Restaurant
            key={restaurant.name + '_' + i}
            restaurant={restaurantInfo}
            isContentsLoading={isContentsLoading}
          />
        );
      })}
    </VenueCardContainer>
  );
};

export default Restaurants;
