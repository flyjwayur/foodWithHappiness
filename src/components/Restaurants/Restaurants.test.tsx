import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Restaurants from './Restaurants';
import 'jest-styled-components';

afterEach(cleanup);

describe('<Restaurants />', () => {
  const dummyRestaurtant = {
    blurhash: 'UUKJMXv|x]oz0gtRM{V@AHRQwvxZXSs9s;o0',
    city: 'Helsinki',
    currency: 'EUR',
    delivery_price: 390,
    description: 'Asenneburgeri',
    image:
      'https://prod-wolt-venue-images-cdn.wolt.com/5b348b31fe8992000bbec771/2be8c7738b220df2f9a0974da5c90d90',
    location: [24.941325187683105, 60.169938852212965],
    name: 'Social Burgerjoint Citycenter',
    online: false,
    tags: ['hamburger', 'fries']
  };

  const testData = [
    { ...dummyRestaurtant, name: 'Social Burgerjoint Citycenter' },
    {
      ...dummyRestaurtant,
      name: 'Momotoko Citycenter'
    },
    {
      ...dummyRestaurtant,
      name: "Luckiefun's Kaivokatu"
    },
    {
      ...dummyRestaurtant,
      name: "Fafa's Sokos"
    }
  ];

  test('Display lists of restaurants', async () => {
    // Arrange
    const isContentsLoading = false;

    const { findByTestId } = render(
      <Restaurants restaurantsList={testData} isContentsLoading={isContentsLoading} />
    );

    const restaurants = await findByTestId('restaurant-list');

    // Act
    // Assert
    expect(restaurants).toMatchSnapshot();
  });
});
