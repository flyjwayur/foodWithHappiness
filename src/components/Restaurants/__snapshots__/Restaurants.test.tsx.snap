// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Restaurants /> Display lists of restaurants 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 20rem;
  border-radius: 8px;
  box-shadow: 0px 0px 10px 0px #aaa;
}

.c1 > img {
  width: 100%;
  height: 12rem;
  object-fit: cover;
}

.c2 {
  border-radius: 8px 8px 0 0;
  width: 210px;
  height: 158px;
}

.c3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  padding: 0.875rem 1rem;
  padding-bottom: 1rem;
  background-color: #eff1f3;
}

.c4 {
  font-weight: 500;
  font-size: 1.125rem;
  color: #3c3d40;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c5 {
  font-weight: 400;
  font-size: 1rem;
  color: #868780;
  margin-top: 2px;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c6 {
  border-top: 1px dashed #e4e4e4;
  font-size: 12px;
  color: #868789;
  padding: 0.5rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  font-size: 0.8rem;
  line-height: 1.3rem;
}

.c8 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  background-color: #009de0;
  color: #ffffff;
  text-align: center;
  font-size: 0.8rem;
  line-height: 1.3rem;
  border-radius: 5px;
  padding: 0.3rem 0.5rem;
  margin: 0.3rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin: 0 -0.5em;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 > div {
  width: calc(33% - 1.065rem);
  margin: 0 0.65rem;
  margin-bottom: 1rem;
}

@media (max-width:1200px) {
  .c1 {
    min-width: 25rem;
  }
}

@media (max-width:900px) {
  .c1 {
    min-width: 19rem;
  }
}

@media (max-width:1200px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:600px) {
  .c0 > div {
    width: 100%;
  }
}

<div
  class="c0"
  data-testid="restaurant-list"
>
  <div
    class="c1"
    data-testid="restaurant"
  >
    <div
      style="display: inline-block; height: 158px; width: 210px; position: relative;"
    >
      <canvas
        height="32"
        punch="1"
        style="position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px; width: 100%; height: 100%;"
        width="32"
      />
    </div>
    <img
      alt="Social Burgerjoint Citycenter_image"
      class="c2"
      src="https://prod-wolt-venue-images-cdn.wolt.com/5b348b31fe8992000bbec771/2be8c7738b220df2f9a0974da5c90d90"
      style="display: none;"
    />
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        Social Burgerjoint Citycenter
      </div>
      <div
        class="c5"
      >
        Asenneburgeri
      </div>
    </div>
    <div
      class="c6"
    >
      <span
        class="c7"
      >
         Delivery 
        3.90
         €
      </span>
      <div
        class=""
      >
        <span
          class="c8"
        >
          hamburger
        </span>
        <span
          class="c8"
        >
          fries
        </span>
      </div>
    </div>
  </div>
  <div
    class="c1"
    data-testid="restaurant"
  >
    <div
      style="display: inline-block; height: 158px; width: 210px; position: relative;"
    >
      <canvas
        height="32"
        punch="1"
        style="position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px; width: 100%; height: 100%;"
        width="32"
      />
    </div>
    <img
      alt="Momotoko Citycenter_image"
      class="c2"
      src="https://prod-wolt-venue-images-cdn.wolt.com/5b348b31fe8992000bbec771/2be8c7738b220df2f9a0974da5c90d90"
      style="display: none;"
    />
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        Momotoko Citycenter
      </div>
      <div
        class="c5"
      >
        Asenneburgeri
      </div>
    </div>
    <div
      class="c6"
    >
      <span
        class="c7"
      >
         Delivery 
        3.90
         €
      </span>
      <div
        class=""
      >
        <span
          class="c8"
        >
          hamburger
        </span>
        <span
          class="c8"
        >
          fries
        </span>
      </div>
    </div>
  </div>
  <div
    class="c1"
    data-testid="restaurant"
  >
    <div
      style="display: inline-block; height: 158px; width: 210px; position: relative;"
    >
      <canvas
        height="32"
        punch="1"
        style="position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px; width: 100%; height: 100%;"
        width="32"
      />
    </div>
    <img
      alt="Luckiefun's Kaivokatu_image"
      class="c2"
      src="https://prod-wolt-venue-images-cdn.wolt.com/5b348b31fe8992000bbec771/2be8c7738b220df2f9a0974da5c90d90"
      style="display: none;"
    />
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        Luckiefun's Kaivokatu
      </div>
      <div
        class="c5"
      >
        Asenneburgeri
      </div>
    </div>
    <div
      class="c6"
    >
      <span
        class="c7"
      >
         Delivery 
        3.90
         €
      </span>
      <div
        class=""
      >
        <span
          class="c8"
        >
          hamburger
        </span>
        <span
          class="c8"
        >
          fries
        </span>
      </div>
    </div>
  </div>
  <div
    class="c1"
    data-testid="restaurant"
  >
    <div
      style="display: inline-block; height: 158px; width: 210px; position: relative;"
    >
      <canvas
        height="32"
        punch="1"
        style="position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px; width: 100%; height: 100%;"
        width="32"
      />
    </div>
    <img
      alt="Fafa's Sokos_image"
      class="c2"
      src="https://prod-wolt-venue-images-cdn.wolt.com/5b348b31fe8992000bbec771/2be8c7738b220df2f9a0974da5c90d90"
      style="display: none;"
    />
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        Fafa's Sokos
      </div>
      <div
        class="c5"
      >
        Asenneburgeri
      </div>
    </div>
    <div
      class="c6"
    >
      <span
        class="c7"
      >
         Delivery 
        3.90
         €
      </span>
      <div
        class=""
      >
        <span
          class="c8"
        >
          hamburger
        </span>
        <span
          class="c8"
        >
          fries
        </span>
      </div>
    </div>
  </div>
</div>
`;
